percona/controller/pgcluster/controller.go

```diff
@@ -535,6 +539,47 @@ func (r *PGClusterReconciler) reconcileCustomExtensions(cr *v2.PerconaPGCluster)
 		extensionKeys = append(extensionKeys, key)
 	}
 
+	if cr.CompareVersion("2.6.0") >= 0 {
+		var extensionKeysForDeletion []string
+		var extensionsKeysAnnotations []string
+
+		if cr.Spec.Metadata.Annotations != nil {
```

Comment:
You don't need to check if the map is nil if you only perform a get operation, you won't get a panic and `ok` will be false.